ports/mimxrt10xx/common-hal/busio/SPI.c

```diff
@@ -35,46 +34,10 @@
 
 #include <stdio.h>
 
-//bool never_reset_sercoms[SERCOM_INST_NUM];
-//
-//void never_reset_sercom(Sercom* sercom) {
-//    // Reset all SERCOMs except the ones being used by on-board devices.
-//    Sercom *sercom_instances[SERCOM_INST_NUM] = SERCOM_INSTS;
-//    for (int i = 0; i < SERCOM_INST_NUM; i++) {
-//        if (sercom_instances[i] == sercom) {
-//            never_reset_sercoms[i] = true;
-//            break;
-//        }
-//    }
-//}
-//
-//void allow_reset_sercom(Sercom* sercom) {
-//    // Reset all SERCOMs except the ones being used by on-board devices.
-//    Sercom *sercom_instances[SERCOM_INST_NUM] = SERCOM_INSTS;
-//    for (int i = 0; i < SERCOM_INST_NUM; i++) {
-//        if (sercom_instances[i] == sercom) {
-//            never_reset_sercoms[i] = false;
-//            break;
-//        }
-//    }
-//}
-//
-//void reset_sercoms(void) {
-//    // Reset all SERCOMs except the ones being used by on-board devices.
-//    Sercom *sercom_instances[SERCOM_INST_NUM] = SERCOM_INSTS;
-//    for (int i = 0; i < SERCOM_INST_NUM; i++) {
-//        if (never_reset_sercoms[i]) {
-//            continue;
-//        }
-//    #ifdef MICROPY_HW_APA102_SERCOM
-//        if (sercom_instances[i] == MICROPY_HW_APA102_SERCOM) {
-//            continue;
-//        }
-//    #endif
-//        // SWRST is same for all modes of SERCOMs.
-//        sercom_instances[i]->SPI.CTRLA.bit.SWRST = 1;
-//    }
-//}
+//arrays use 0 based numbering: SPI1 is stored at index 0
+#define MAX_SPI 4
+STATIC bool reserved_spi[MAX_SPI];
+STATIC bool never_reset_spi[MAX_SPI];
```

Comment:
A bit inconsistent using STATIC here and next line uses static, but minor detail.